src/careamics/models/lvae/lvae.py

```diff
@@ -494,29 +447,28 @@ def create_likelihood_module(self):
         This method defines the likelihood module for the current LVAE model.
         The existing likelihood modules are `GaussianLikelihood` and `NoiseModelLikelihood`.
         """
-        self.likelihood_gm = GaussianLikelihood(
-            self.decoder_n_filters,
-            self.target_ch,
-            predict_logvar=self.predict_logvar,
-            logvar_lowerbound=self.logvar_lowerbound,
-            conv2d_bias=self.topdown_conv2d_bias,
-        )
-
-        self.likelihood_NM = None
         if self.enable_noise_model:
+            raise NotImplementedError("Noise model not implemented yet.")
             self.likelihood_NM = NoiseModelLikelihood(
                 self.decoder_n_filters,
                 self.target_ch,
                 self.data_mean,
                 self.data_std,
                 self.noiseModel,
             )
-        if self.loss_type == LossType.DenoiSplitMuSplit or self.likelihood_NM is None:
+        else:
+            self.likelihood_gm = GaussianLikelihood(
+                self.decoder_n_filters,
+                self.target_ch,
+                predict_logvar=self.predict_logvar,
+                logvar_lowerbound=self.logvar_lowerbound,
+                conv2d_bias=self.topdown_conv2d_bias,
+            )
```

Comment:
When is `self.likelihood_gm` used? as far as I can see it doesn't seem to be used by any of the`LVAE` class' methods. If it is only used by external functions maybe it can be removed from the `LVAE` class and passed to those functions explicitly.